tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -335,21 +335,23 @@ def __init__(self, data, manager=None):
 
     def process(self, resources, event=None):
         s = self.manager.get_session()
+        definition_ids = None
 
         # Translate definitions display names into ids
-        policyClient = s.client("azure.mgmt.resource.policy.PolicyClient")
-        definitions = [d for d in policyClient.policy_definitions.list()]
-        definition_ids = [d.id.lower() for d in definitions
-                          if self.definitions is None or
-                          d.display_name in self.definitions or
-                          d.name in self.definitions]
+        if self.definitions:
+            policyClient = s.client("azure.mgmt.resource.policy.PolicyClient")
+            definitions = [d for d in policyClient.policy_definitions.list()]
+            definition_ids = [d.id.lower() for d in definitions
+                              if self.definitions is None or
```

Comment:
Since we are checking `if self.definitions`, can we remove `if self.definitions is None` this check inside here?

Reply:
Yep I missed that, tx.